### PR TITLE
P2 and P3 cases - Update size for support both gen1 and gen2 images on Azure.

### PR DESCRIPTION
**Pull request by LiliDeng**
Test results -
VERIFY-DPDK-NFF-GO and VERIFY-DPDK-PRIMARY-SECONDARY-PROCESSES abort for other reason, not related with deployment error.

```
[LISAv2 Test Results Summary]
Test Run On           : 04/07/2020 09:24:43
ARM Image Under Test  : canonical : ubuntuserver : 18_04-lts-gen2 : latest
Initial Kernel Version: 5.0.0-1035-azure
Final Kernel Version  : 5.0.0-1035-azure
Total Test Cases      : 9 (7 Passed, 0 Failed, 2 Aborted, 0 Skipped)
Total Time (dd:hh:mm) : 0:1:50

   ID TestArea             TestCaseName                                                                TestResult TestDuration(in minutes) 
-------------------------------------------------------------------------------------------------------------------------------------------
    1 NESTED               NESTED-KVM-NTTTCP-PRIVATE-BRIDGE                                                  PASS                26.09 
    2 DPDK                 VERIFY-DPDK-OVS                                                                   PASS                11.53 
    3 NESTED               AZURE-NESTED-KVM-NTTTCP-DIFFERENT-L1-NAT                                          PASS                27.23
    4 NESTED               AZURE-NESTED-KVM-NETPERF-PPS                                                      PASS                12.57
    5 STRESS               STRESSTEST-WEB-PARALLEL-ACCESS                                                    PASS                 11.6 
    6 DPDK                 VERIFY-DPDK-NFF-GO                                                             ABORTED                 7.91 
    7 DPDK                 VERIFY-DPDK-VPP                                                                   PASS                15.39 
    8 DPDK                 VERIFY-DPDK-PRIMARY-SECONDARY-PROCESSES                                        ABORTED                13.56 
    9 GOLANG               PERF-GOLANG-BENCHMARK                                                             PASS                 7.37 

[LISAv2 Test Results Summary]
Test Run On           : 04/07/2020 12:47:03
VHD Under Test        : ubuntu_18.04.1_gen1_gen2.vhdx
Initial Kernel Version: 4.15.0-38-generic
Final Kernel Version  : 5.0.0-1036-azure
Total Test Cases      : 2 (2 Passed, 0 Failed, 0 Aborted, 0 Skipped)
Total Time (dd:hh:mm) : 0:0:32

   ID TestArea             TestCaseName                                                                TestResult TestDuration(in minutes) 
-------------------------------------------------------------------------------------------------------------------------------------------
    1 STRESS               STRESSTEST-WEB-PARALLEL-ACCESS                                                    PASS                16.29 
    2 GOLANG               PERF-GOLANG-BENCHMARK                                                             PASS                 2.78 
```

### CODE REVIEW

File: XML/TestCases/FunctionalTests-DPDK.xml [hunk 1]
```diff
@@ -80,7 +80,7 @@
         <AdditionalHWConfig>
             <Networking>SRIOV</Networking>
         </AdditionalHWConfig>
-        <OverrideVMSize>Standard_D16_v3</OverrideVMSize>
+        <OverrideVMSize>Standard_D16s_v3</OverrideVMSize>
```
Comment (juhlee-microsoft) [inline XML/TestCases/FunctionalTests-DPDK.xml:83]: Can you switch to smaller size?

File: XML/TestCases/FunctionalTests-DPDK.xml [hunk 2]
```diff
@@ -117,7 +117,7 @@
         <testName>VERIFY-DPDK-NFF-GO</testName>
         <testScript>VERIFY-DPDK-NFF-GO.ps1</testScript>
         <setupType>OneVM3NIC</setupType>
-        <OverrideVMSize>Standard_D16_v3</OverrideVMSize>
+        <OverrideVMSize>Standard_D16s_v3</OverrideVMSize>
```
Comment (juhlee-microsoft) [inline XML/TestCases/FunctionalTests-DPDK.xml:120]: can you switch to smaller size?

File: XML/TestCases/FunctionalTests-DPDK.xml [hunk 3]
```diff
@@ -138,7 +138,7 @@
         <testName>VERIFY-DPDK-VPP</testName>
         <testScript>VERIFY-DPDK-VPP.ps1</testScript>
         <setupType>OneVM3NIC</setupType>
-        <OverrideVMSize>Standard_D16_v3</OverrideVMSize>
+        <OverrideVMSize>Standard_D16s_v3</OverrideVMSize>
```
Comment (juhlee-microsoft) [inline XML/TestCases/FunctionalTests-DPDK.xml:141]: @abhimarathe Can you switch to smaller size?

File: XML/TestCases/FunctionalTests-DPDK.xml [hunk 4]
```diff
@@ -157,7 +157,7 @@
         <testName>VERIFY-DPDK-PRIMARY-SECONDARY-PROCESSES</testName>
         <testScript>VERIFY-DPDK-PRIMARY-SECONDARY-PROCESSES.ps1</testScript>
         <setupType>OneVM3NIC</setupType>
-        <OverrideVMSize>Standard_D16_v3</OverrideVMSize>
+        <OverrideVMSize>Standard_D16s_v3</OverrideVMSize>
```
Comment (juhlee-microsoft) [inline XML/TestCases/FunctionalTests-DPDK.xml:160]: @abhimarathe Can you switch to smaller size?

File: XML/TestCases/NestedVmTests.xml [hunk 5]
```diff
@@ -24,7 +24,7 @@
         <testScript>NESTED-KVM-NTTTCP-PRIVATE-BRIDGE.ps1</testScript>
         <files>.\Testscripts\Linux\utils.sh,.\Testscripts\Linux\nested_vm_utils.sh,.\Testscripts\Linux\nested_kvm_ntttcp_private_bridge.sh,.\Testscripts\Linux\enableRoot.sh,.\Testscripts\Linux\enablePasswordLessRoot.sh,.\Testscripts\Linux\perf_ntttcp.sh</files>
         <setupType>OneVM</setupType>
-        <OverrideVMSize>Standard_E16_v3</OverrideVMSize>
+        <OverrideVMSize>Standard_E16s_v3</OverrideVMSize>
```
Comment (juhlee-microsoft) [inline XML/TestCases/NestedVmTests.xml:27]: @LiliDeng Can you switch to smaller size?

File: XML/TestCases/NestedVmTests.xml [hunk 6]
```diff
@@ -294,7 +294,7 @@
         <testScript>NESTED-KVM-NTTTCP-DIFFERENT-L1-PUBLIC-BRIDGE.ps1</testScript>
         <files>.\Testscripts\Linux\utils.sh,.\Testscripts\Linux\nested_vm_utils.sh,.\Testscripts\Linux\nested_kvm_ntttcp_different_l1_nat.sh,.\Testscripts\Linux\enableRoot.sh,.\Testscripts\Linux\enablePasswordLessRoot.sh,.\Testscripts\Linux\perf_ntttcp.sh,.\Testscripts\Linux\nat_qemu_ifup.sh,</files>
         <setupType>TwoVM2NIC</setupType>
-        <OverrideVMSize>Standard_E16_v3</OverrideVMSize>
+        <OverrideVMSize>Standard_E16s_v3</OverrideVMSize>
```
Comment (juhlee-microsoft) [inline XML/TestCases/NestedVmTests.xml:297]: @LiliDeng Can you switch to smaller size?

File: XML/TestCases/NestedVmTests.xml [hunk 7]
```diff
@@ -492,7 +492,7 @@
         <testScript>NESTED-KVM-NETPERF-PPS.ps1</testScript>
         <files>.\Testscripts\Linux\utils.sh,.\Testscripts\Linux\nested_vm_utils.sh,.\Testscripts\Linux\nested_kvm_netperf_pps_nat.sh,.\Testscripts\Linux\enableRoot.sh,.\Testscripts\Linux\enablePasswordLessRoot.sh,.\Testscripts\Linux\perf_netperf.sh,.\Testscripts\Linux\nat_qemu_ifup.sh</files>
         <setupType>TwoVM2NIC</setupType>
-        <OverrideVMSize>Standard_E16_v3</OverrideVMSize>
+        <OverrideVMSize>Standard_E16s_v3</OverrideVMSize>
```
Comment (juhlee-microsoft) [inline XML/TestCases/NestedVmTests.xml:495]: @LiliDeng Can you switch to smaller size?

File: XML/TestCases/PerformanceTests.xml [hunk 8]
```diff
@@ -629,7 +629,7 @@
         <testName>PERF-GOLANG-BENCHMARK</testName>
         <TestScript>PERF-GOLANG-BENCHMARK.ps1</TestScript>
         <setupType>OneVM</setupType>
-        <OverrideVMSize>Standard_D13_v2</OverrideVMSize>
+        <OverrideVMSize>Standard_DS13_v2</OverrideVMSize>
```
Comment (juhlee-microsoft) [inline XML/TestCases/PerformanceTests.xml:632]: @LiliDeng Can you switch to smaller size?

File: XML/TestCases/PerformanceTests.xml [hunk 9]
```diff
@@ -646,7 +646,7 @@
         <preTestScript>GPU-DRIVER-INSTALL.ps1</preTestScript>
         <testScript>GPU-TENSORFLOW.ps1</testScript>
         <setupType>OneVM</setupType>
-        <OverrideVMSize>Standard_NC6</OverrideVMSize>
+        <OverrideVMSize>Standard_NC6s_v2</OverrideVMSize>
```
Comment (juhlee-microsoft) [inline XML/TestCases/PerformanceTests.xml:649]: @LiliDeng Can you switch to smaller size?

**Comment by juhlee-microsoft** [comment]
Can you switch to smaller size? @abhimarathe @LiliDeng

File: XML/TestCases/FunctionalTests-DPDK.xml (hunk 1 above)
Comment (abhimarathe) [inline XML/TestCases/FunctionalTests-DPDK.xml:83]: Why we are moving it from D16 to D16s? Is it requirement by gen2 vms or due to reducing usage?

**Review by abhimarathe — COMMENTED** [review]
By looking at the functionality of OVS, VPP, Nff-go and Primary-secondary testcases, we can reduce the vm sizes to D4s. Need to test it. I'll do the testing and report back.

**Comment by abhimarathe** [comment]
These all testcases are using 3 NIC configuration. I tried changing vm size to D4 but deployment failed as maximum 2 SRIOV NICs can be attached to D4s_v3/D4_v3 size VMs.
I tried looking for VM size vs max SRIOV NIC allowed doc. couldn't find one.
AN doc here (https://docs.microsoft.com/en-us/azure/virtual-network/create-vm-accelerated-networking-cli) says Dsv3 size vms runs with 4 or more SRIOV Nics but it isn't true in this case.

**Comment by LiliDeng** [comment]
> Can you switch to smaller size? @abhimarathe @LiliDeng

I suggest that we cover this in another PR, this is for support both gen1 and gen2 image.

File: XML/TestCases/FunctionalTests-DPDK.xml (hunk 1 above)
Comment (LiliDeng) [reply XML/TestCases/FunctionalTests-DPDK.xml:83]: for gen2 image

**Comment by LiliDeng** [comment]
@abhimarathe @juhlee-microsoft Let's focus on this simply, I use this PR to make sure the size support both gen1 and gen2 images; for reduce the size, save cost, let's create another PR for it.

**Review by abhimarathe — APPROVED** [review]

**Pull request merged**